source/extensions/filters/http/istio_stats/istio_stats.cc

```diff
@@ -0,0 +1,669 @@
+        const auto upstream_info = info.upstreamInfo();
+        const Ssl::ConnectionInfoConstSharedPtr ssl_info =
+            upstream_info ? upstream_info->upstreamSslConnection() : nullptr;
+        tags_.push_back(
+            {context_.source_principal_,
+             ssl_info && !ssl_info->uriSanLocalCertificate().empty()
+                 ? config_->resolve(ssl_info->uriSanLocalCertificate()[0])
+                 : context_.unknown_});
+        tags_.push_back({context_.source_app_, context_.app_name_});
+        tags_.push_back({context_.source_version_, context_.app_version_});
+        tags_.push_back({context_.source_cluster_, context_.cluster_name_});
+        tags_.push_back({context_.destination_workload_,
+                         peer ? config_->resolve(peer->workload_name_)
+                              : context_.unknown_});
+        tags_.push_back({context_.destination_workload_namespace_,
+                         peer ? config_->resolve(peer->namespace_name_)
+                              : context_.unknown_});
+        tags_.push_back(
+            {context_.destination_principal_,
+             ssl_info && !ssl_info->uriSanPeerCertificate().empty()
+                 ? config_->resolve(ssl_info->uriSanPeerCertificate()[0])
+                 : context_.unknown_});
+        tags_.push_back(
+            {context_.destination_app_,
+             peer ? config_->resolve(peer->app_name_) : context_.unknown_});
+        tags_.push_back(
+            {context_.destination_version_,
+             peer ? config_->resolve(peer->app_version_) : context_.unknown_});
+        tags_.push_back({context_.destination_service_,
+                         service_host.empty()
+                             ? context_.unknown_
+                             : config_->resolve(service_host)});
+        tags_.push_back({context_.destination_canonical_service_,
+                         peer ? config_->resolve(peer->canonical_name_)
+                              : context_.unknown_});
+        tags_.push_back({context_.destination_canonical_revision_,
+                         peer ? config_->resolve(peer->canonical_revision_)
+                              : context_.latest_});
+        tags_.push_back({context_.destination_service_name_,
+                         service_host_name.empty()
+                             ? context_.unknown_
+                             : config_->resolve(service_host_name)});
+        tags_.push_back({context_.destination_service_namespace_,
+                         peer ? config_->resolve(peer->namespace_name_)
+                              : context_.unknown_});
+        tags_.push_back(
+            {context_.destination_cluster_,
+             peer ? config_->resolve(peer->cluster_name_) : context_.unknown_});
+        break;
+      }
+      default:
+        break;
+    }
+  }
+
+  ConfigSharedPtr config_;
+  Context& context_;
+  Stats::StatNameTagVector tags_;
```

Comment:
I don't know what the default vector capacity is, maybe should initialize this vector with the worst-case scenario? to avoid needing to grow the vector in most cases

Reply:
Good idea, reserved it for maximum 25 (number of HTTP dimensions).